File: app/src/main/java/com/slotmachine/ocr/mic/BaseActivity.java
package com.slotmachine.ocr.mic;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.analytics.FirebaseAnalytics;

public class BaseActivity extends AppCompatActivity {

    private FirebaseAnalytics mFirebaseAnalytics;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mFirebaseAnalytics = FirebaseAnalytics.getInstance(this);
    }

    protected void onEvent(String eventName, String param) {
        Bundle bundle = new Bundle();
        bundle.putString("param", param);
        onEvent(eventName, bundle);
    }

    private void onEvent(String eventName, Bundle bundle) {
        mFirebaseAnalytics.logEvent(eventName, bundle);
    }
}

File: app/src/main/java/com/slotmachine/ocr/mic/BaseActivity.kt
package com.slotmachine.ocr.mic

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.analytics.ktx.analytics
import com.google.firebase.ktx.Firebase

open class BaseActivity : AppCompatActivity() {

    private lateinit var mFirebaseAnalytics: FirebaseAnalytics

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mFirebaseAnalytics = Firebase.analytics
    }

    protected fun onEvent(eventName: String, param: String?) {
        val bundle = Bundle()
        bundle.putString("param", param)
        onEvent(eventName, bundle)
    }

    private fun onEvent(eventName: String, bundle: Bundle) {
        mFirebaseAnalytics.logEvent(eventName, bundle)
    }

}